### DISCUSSION TITLE
Actions secrets and variables on enterprise level

### DISCUSSION THREAD

**Discussion by Timmmy-nlb** [community/community#77293] [Enterprise]
### Select Topic Area

Product Feedback

### Body

What I miss and would like to see as a new function would be secrets and variables at enterprise level, which are then inherited by the organizations.

Especially for license keys, shared tokens or general variables, it would be great if you didn't have to create them manually in each org.

I would imagine that it would be possible to restrict the individual entries as to which orgs they are inherited by, as with the runner groups.

**Comment by CheetahCodes21** [comment]
Having the ability to define secrets and variables at the enterprise level and inherit them across organizations would be a valuable addition to GitHub Enterprise. This would streamline the process of managing sensitive information and shared configurations, reducing the need for manual duplication and ensuring consistency across the organization.

Here's a more detailed breakdown of the proposed functionality:

1. **Enterprise-level secrets and variables:** Create a dedicated section within the enterprise settings to manage secrets and variables that apply to all organizations. These secrets and variables could be defined using a key-value pair format, similar to the existing secrets and variables management at the organization and repository levels.

2. **Inheritance and restrictions:** Implement a mechanism to inherit enterprise-level secrets and variables to specific organizations. This could be achieved through a hierarchical inheritance model or by allowing administrators to explicitly select which organizations should inherit specific secrets and variables.

3. **Access control:** Maintain granular access control for enterprise-level secrets and variables. This ensures that only authorized individuals can view, create, update, or delete these entities.

4. **Consistency and security:** By centralizing the management of sensitive information and shared configurations, organizations can achieve greater consistency and security across their GitHub Enterprise environment. This reduces the risk of misconfigurations, unauthorized access, and data breaches.

5. **Reduced administrative overhead:** Eliminating the need to manually create and manage secrets and variables in each organization simplifies administrative tasks and reduces the overall workload for IT teams.

6. **Improved collaboration:** Enterprise-level secrets and variables can facilitate seamless collaboration across organizations, ensuring that teams have access to the necessary information and configurations to work effectively.

Overall, introducing enterprise-level secrets and variables would enhance the manageability, security, and collaboration within GitHub Enterprise organizations.

**Reply by Akash1134** [reply]
Hi @Timmmy-nlb , 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Reply by Timmmy-nlb** [reply]
Hello @Akash1134,
I'm a bit confused about the address to me.
And I honestly don't understand it in the context of a functional request...

**Reply by Timmmy-nlb** [reply]
Hello @Akash1134 ,
what does it look like now?
Are there any questions to confirm the usefulness of this function or has this request been accepted as valid?

**Reply by Timmmy-nlb** [reply]
@Akash1134 Hello? No response?

**Comment by jasontempleman-eaton** [comment]
I don't see any movement on this and it is a needed feature.

**Comment by razorsk8jz** [comment]
This would be a really nice to have

**Comment by lee5i3** [comment]
Having the ability to define secrets and variables at the enterprise level and have them automatically inherited across organizations would be incredibly valuable. Right now, managing shared variables across multiple organizations requires a lot of manual duplication. Not only is that time-consuming, but it also increases the risk of inconsistencies or outdated values slipping through.

If we could centrally manage secrets and variables at the enterprise level, it would make it much easier to keep everything aligned across all organizations. It would reduce administrative overhead, simplify onboarding new repos or orgs, and give us more confidence that sensitive information is being handled consistently and securely everywhere.

**Comment by T-me-813** [comment]
I also think having enterprise level control would be ideal for numerous scenarios.

**Comment by zeeket** [comment]
A use for Enterprise level secrets and variables would in actions or reusable workflows that are used across organisations in the enterprise.
Please implement this. Along with Team-based secrets and variables, this would be very much helpful and needed.
I already like how organization level secrets can be scoped to certain repositories. Please continue the work needed to make Actions, Secrets and Variables a scalable solution.

**Comment by andycohenbbc** [comment]
Another request for this feature. It is a gap at the Enterprise level not to be able to inherit secrets across your Github organisations.
